easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -502,8 +511,11 @@ def include_defined_parameters(keyset):
             for group in keyset:
                 printed = False
                 for key in group:
-                    if self[key] != default_values[key]:
-                        ebtxt.append("%s = %s" % (key, quote_str(self[key], escape_newline=True)))
+                    val = self[key]
+                    if val != default_values[key]:
+                        if key not in exclude_keys:
+                            val = replace_templates(val, templ_const, templ_val)
```

Comment:
add some logging here, so we can debug replacements that shouldn't be happening:

```python
self.log.debug("Original value before replacing matching template values: %s", val)
val = replace_templates(val, templ_const, templ_val)
self.log.debug("New value after replacing matching template values: %s", val)
```